File: src/codes/tight-coupling/py/ksrc.py
class InventoryService:
    def reduce_stock(self, product_id: str, amount: int):
        print(f"在庫を減らしました: {product_id} × {amount}")

class OrderService:
    def __init__(self):
        self.inventory = InventoryService()

    def place_order(self, product_id: str, quantity: int):
        print(f"注文受付: {product_id} × {quantity}")
        self.inventory.reduce_stock(product_id, quantity)

File: src/codes/tight-coupling/php/ksrc.php
<?php

class InventoryService {
  public function reduceStock(string $productId, int $amount): void {
    echo "在庫を減らしました: {$productId} × {$amount}\n";
  }
}

class OrderService {
  private InventoryService $inventory;

  public function __construct() {
    $this->inventory = new InventoryService();
  }

  public function placeOrder(string $productId, int $quantity): void {
    echo "注文受付: {$productId} × {$quantity}\n";
    $this->inventory->reduceStock($productId, $quantity);
  }
}
